drivers/wireless/bluetooth/bt_rpmsghci.c

```diff
@@ -0,0 +1,535 @@
+  int                      ret = OK;
+
+  memset(&cookie, 0, sizeof(struct rpmsghci_cookie_s));
+  nxsem_init(&cookie.sem, 0, 0);
+
+  msg->command = command;
+  msg->result  = -ENXIO;
+  msg->cookie  = (uintptr_t)&cookie;
+
+  ret = rpmsg_send_nocopy(&priv->ept, msg, len);
+  if (ret < 0)
+    {
+      goto errout;
+    }
+
+  ret = rpmsg_wait(&priv->ept, &cookie.sem);
+  if (ret >= 0)
+    {
+      ret = msg->result;
+    }
+
+errout:
+  nxsem_destroy(&cookie.sem);
+  return ret;
+}
+
+/****************************************************************************
+ * Name: bt_rpmsghci_open
+ *
+ * Description:
+ *   Rpmsg-HCI open operation
+ *
+ * Parameters:
+ *   btdev  - the bt instance
+ *
+ * Returned Values:
+ *   OK on success; A negated errno value is returned on any failure.
+ *
+ ****************************************************************************/
+
+static int bt_rpmsghci_open(struct bt_driver_s *btdev)
+{
+  FAR struct rpmsghci_s      *priv = NULL;
+  FAR struct rpmsghci_open_s *msg  = NULL;
+  uint32_t                    space = 0;
+  int                         ret  = OK;
+
+  /* Get RPMSG-HCI data */
+
+  DEBUGASSERT(btdev != NULL);
+  priv = btdev->rpmsg;
+
+  /* Perform the rpmsg write */
+
+  msg = rpmsghci_get_tx_payload_buffer(priv, &space);
+  if (msg == NULL)
+    {
+      ret = -ENOMEM;
+      goto errout;
+    }
+
+  ret = rpmsghci_send(priv, RPMSGHCI_OPEN, &msg->header,
+                      sizeof(struct rpmsghci_open_s));
+errout:
+  return ret;
+}
+
+/****************************************************************************
+ * Name: bt_rpmsghci_close
+ *
+ * Description:
+ *   Rpmsg-HCI close operation
+ *
+ * Parameters:
+ *   btdev  - the bt instance
+ *
+ * Returned Values:
+ *   OK on success; A negated errno value is returned on any failure.
+ *
+ ****************************************************************************/
+
+static void bt_rpmsghci_close(struct bt_driver_s *btdev)
+{
+  FAR struct rpmsghci_s       *priv  = NULL;
+  FAR struct rpmsghci_close_s *msg   = NULL;
+  uint32_t                     space = 0;
+
+  /* Get RPMSG-HCI data */
+
+  DEBUGASSERT(btdev != NULL);
+  priv = btdev->rpmsg;
+
+  /* Perform the rpmsg write */
+
+  msg = rpmsghci_get_tx_payload_buffer(priv, &space);
+  if (msg == NULL)
+    {
+      return;
+    }
+
+  rpmsghci_send(priv, RPMSGHCI_CLOSE, &msg->header,
+                sizeof(struct rpmsghci_close_s));
+}
+
+/****************************************************************************
+ * Name: bt_rpmsghci_send
+ *
+ * Description:
+ *   Rpmsg-HCI send operation
+ *
+ * Parameters:
+ *   btdev  - the bt instance
+ *   type   - bt command type
+ *   data   - bt command data
+ *   len    - bt command data length
+ *
+ * Returned Values:
+ *   OK on success; A negated errno value is returned on any failure.
+ *
+ ****************************************************************************/
+
+static int bt_rpmsghci_send(struct bt_driver_s *btdev,
```

Comment:
bt_rpmsghci_->rpmsghci_

Reply:
this is bt driver ops, do not confuse with `rpmsghci_send` which is rpmsg data send operation.

Reply:
done for all bt driver ops